compass/src/components/Application/ApplicationDetails/ConnectApplicationModal/ConnectApplication.js

```diff
@@ -39,7 +40,13 @@ const FormEntry = ({ caption, name, value }) => (
 );
 
 export default function ConnectApplicationModal({ applicationId }) {
-  const [connectApplicationMutation] = useMutation(CONNECT_APPLICATION);
+  const [connectApplicationMutation] = useMutation(CONNECT_APPLICATION, {
+    context: {
+      headers: {
+        client_user: getUserId(),
```

Comment:
The `client_user` property is currently hard-coded here. This causes an unquestionable double maintenance between here and where the Director searches for this header: https://github.com/kyma-incubator/compass/blob/1e543bc032f674b8b2005039ddc37e220d3c058b/components/director/internal/authenticator/middleware.go#L29

I suggest introducing a central configuration property in the charts(maybe something like `director.userIDKey: client_user` which defines this header key (actually the value would be set in the mp-config repo). Then this configuration should be plugged in the director env and the constant removed there. In the Cockpit the value should somehow be retrieved from the env and used here. Also the virtual service resources could use the configuration as well instead of hardcoding the string `client_user` as it is now.

This way we can change the expected header key in a central and single place.